src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/CommonRules/PosBlockSignatureRuleTest.cs

```diff
@@ -306,5 +307,96 @@ public async Task RunAsync_ProofOfWorkBlock_BlockSignatureEmpty_DoesNotThrowExce
 
             await this.consensusRules.RegisterRule<PosBlockSignatureRule>().RunAsync(this.ruleContext);
         }
+
+        /// <summary>
+        /// This helper creates a cold coin staking block containing a cold coin staking transaction built according to
+        /// the parameters and with valid first input and output. The block signature is created correctly with the
+        /// private key corresponding to the public key.
+        /// </summary>
+        /// <param name="useCompressedKey">Determines whether the second transaction output will include a compressed
+        /// (versus uncompressed) public key.</param>
+        /// <param name="includeSecondPush">Determines whether the second transaction output will include a small integer
+        /// after the public key.</param>
+        /// <param name="expectFailure">Determines whether we expect failure (versus success).</param>
+        private void ProofOfStakeBlock_ColdStakeTestHelper(bool useCompressedKey, bool includeSecondPush, bool expectFailure)
```

Comment:
Do you want to bring this in now?

Reply:
Yes it is a test for the functionality, but possibly we may want to rename it? Do we care?

Reply:
yeah let's rename it, as this is not only for cold staking.